boardgame-ui/src/components/event/EventAddEditGame.vue

```diff
@@ -0,0 +1,314 @@
+
+export default {
+  components: {
+    UserAutocomplete
+  },
+  props: {
+    event: Object,
+    editGame: Object
+  },
+  data() {
+    return {
+      game: null,
+      searchString: '',
+      time: null,
+      minTime: null,
+      players: [],
+      boardGamesLinks: null
+    };
+  },
+  computed: {
+    currentUser() {
+      return this.$store.state.currentUser;
+    },
+    boardGames() {
+      if(!this.boardGamesLinks) {
+        return [];
+      }
+
+      let processedIds = [];
+      let boardGames = [];
+      for(let link of this.boardGamesLinks) {
+        let id = link.id_board_game;
+        if(!processedIds.includes(id)) {
+          boardGames.push(link.provided_board_game);
+          processedIds.push(id);
+        }
+      }
+      return boardGames;
+    },
+    filteredBoardGames() {
+      if(!this.searchString) {
+        return this.boardGames;
+      }
+
+      let str = this.searchString.toLowerCase();
+      return this.boardGames.filter(bg => bg.name.toLowerCase().indexOf(str) >= 0);
+    },
+    allowedRankingMethods() {
+      return [GameRankingMethods.WIN_LOSE, GameRankingMethods.POINTS_HIGHER_BETTER];
+    },
+    ranked() {
+      return this.game.ranking_method === GameRankingMethods.POINTS_HIGHER_BETTER;
+    },
+    users() {
+      return this.event.attendees.map(attendee => attendee.user);
+    },
+    selectedUsersIds() {
+      return this.players.map(({user}) => user ? user.id : 0);
+    }
+  },
+  watch: {
+    'game.ranking_method'(_, old) {
+      if(!old) {
+        return;
+      }
+      this.players.forEach(player => player.score = null); // reinitialize scores
+    },
+    time() {
+      if(this.time < this.minTime) {
+        this.time = this.minTime;
+      }
+    }
+  },
+  methods: {
+    selectBoardGame(option) {
+      this.game.id_board_game = option ? option.id : null;
+    },
+    addPlayer() {
+      this.players.push({user: null, score: null});
+    },
+    removePlayer(idx) {
+      if(this.players.length === 1) {
+        this.$toast.open({
+          message: this.$t('add-edit-game.must-have-at-least-one-player'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+        return;
+      }
+      this.players.splice(idx, 1);
+    },
+    async save() {
+      let result = await this.$validator.validateAll();
+
+      if(!this.game.id_board_game) {
+        this.$validator.errors.add({field: 'boardGame', msg: this.$t('add-edit-game.validation-error.board-game')});
+        result = false;
+      }
+
+      if(!result) {
+        this.$toast.open({
+          message: this.$t('global.invalid-form'),
+          type: 'is-danger',
+          position: 'is-bottom'
+        });
+        return;
+      }
+
+      this.game.duration = this.time.getHours()*60 + this.time.getMinutes();
+      this.game.players = this.players.map(({user, score}) => ({user: user.id, score: Number(score)}));
+
+      try {
+        await this.game.save();
```

Comment:
Note: soon I will implement proper validation on server side for all create/edit requests (this is only done for events right now). Basically you will receive an object giving information about all incorrect fields.